### PR TITLE
[ocm-addons-upgrade-tests-trigger] introduce new integration

### PR DESCRIPTION
**Pull request by maorfr**
https://issues.redhat.com/browse/APPSRE-6756

depends on https://github.com/app-sre/qontract-schemas/pull/363

this integration will trigger jobs in jenkins following an upgrade of an addon for all clusters within an OCM org.

### CODE REVIEW

**Review by chassing — APPROVED** [review]
Quick & dirty :heart: LGTM

**Comment by OliviaHY** [comment]
LGTM

File: reconcile/ocm_addons_upgrade_tests_trigger.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            init_addons=True,
+        )
+
+        ocm = ocm_map[ocm_name]
+        state_updates: dict[str, Optional[str]] = {}
+        for aut in addon_upgrade_tests:
+            addon_name = aut["addon"]["name"]
+            addon_org_version = None
+            should_trigger = True
+            state_key = f"{ocm_name}/{addon_name}"
```
Comment (rporres) [inline reconcile/ocm_addons_upgrade_tests_trigger.py:40]: you may want to take into account @patjlm comment about the state key in https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/55019#note_5540116

**Review by rporres — COMMENTED** [review]
Looks fine for the purpose. I have a doubt about the state key depending on how long we want this to live...

**Review by patjlm — APPROVED** [review]
LGTM. We might need to include sector info in the state, group clusters by sector and trigger job per sector (passing that sector info as a parameter). I agree it can be done later on, but we may have some time of unclear water once sectors are implemented in OCM (the job may need to be triggered manually until we sync sector info down to the state and maybe app-interface?).

**Comment by maorfr** [comment]
the state key can be changed at any point in the future. this would mean that someone could run the integration locally with the trigger part commented out, thus only updating the state.

or, we could just let the jobs be triggered...

**Pull request merged**